ceph-salt-formula/salt/_modules/ceph_salt.py

```diff
@@ -2,13 +2,15 @@
 import json
 
 def get_remote_grain(host, grain):
-    ret = __salt__['cmd.run_all']("ssh -o StrictHostKeyChecking=no root@{} "
+    ret = __salt__['cmd.run_all']("ssh -o StrictHostKeyChecking=no "
+                                  "-i /tmp/ceph-salt-ssh-id_rsa root@{} "
```

Comment:
fyi, mgr/cephadm uses 

https://github.com/ceph/ceph/blob/master/src/pybind/mgr/cephadm/module.py#L63-L69

for its ssh config.